pkcs11/gkm/gkm-data-der.c

```diff
@@ -57,14 +61,168 @@ init_quarks (void)
 
 		QUARK (OID_PKIX1_RSA, "1.2.840.113549.1.1.1");
 		QUARK (OID_PKIX1_DSA, "1.2.840.10040.4.1");
+		QUARK (OID_PKIX1_ECDSA, "1.2.840.10045.2.1");
 		QUARK (OID_PKCS12_PBE_3DES_SHA1, "1.2.840.113549.1.12.1.3");
+		QUARK (OID_ANSI_SECP256R1, "1.2.840.10045.3.1.7");
+		QUARK (OID_ANSI_SECP384R1, "1.3.132.0.34");
+		QUARK (OID_ANSI_SECP521R1, "1.3.132.0.35");
 
 		#undef QUARK
 
 		g_once_init_leave (&quarks_inited, 1);
 	}
 }
 
+const gchar *
+gkm_data_der_oid_to_curve (GQuark oid)
+{
+	if (oid == OID_ANSI_SECP256R1)
+		return "NIST P-256";
+	else if (oid == OID_ANSI_SECP384R1)
+		return "NIST P-384";
+	else if (oid == OID_ANSI_SECP521R1)
+		return "NIST P-521";
+	return NULL;
+}
+
+GQuark
+gkm_data_der_curve_to_oid (gchar *curve)
```

Comment:
`grep`-ing through the code, it is just internal function used only in this file.

I will remove it from the header files and add a comment clarifying it is using `libgcrypt` name of curve as an input name, which should be again created only throughout this code and we should be able to ignore aliases. Will be in a0ec2b20.